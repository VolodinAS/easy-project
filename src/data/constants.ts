export const localStorageTokenKey = 'easy-project.auth-token';