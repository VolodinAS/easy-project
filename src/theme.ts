const theme = {
    color: {
        typeTask: '#F8BFBF',
        typeFeature: '#B7CFFF',
        typeBug: '#FEFBBD',
        black: '#000000',
        lightPurple: '#CFC7D9',
        purple: '#7A52AA',
        lightGray: '#F5F5F5',
        darkGray: '#9C9C9C',
        red: '#EB0C0C',
        white: '#FFFFFF',
        smallText: '#999999',
        opacityGray: '#ACACAC',
        landingColors: {
            mainGray: '#D5E6EA',
            mainBlue: '#3848A9',
            benefitsItemBackground: '#EDD8BE',
            buttonStartBackground: '#E43958',
            footerBackground: '#343F53',
            headerLogo: '#010101',
            headerBackground: '#F1E7DA'
        },
        dashboardColors: {
            overlayButtonCreate: '#ABE0E7',
            overlayButtonYes: '#B4E7AB',
            overlayButtonNo: '#F9CCBE',
        }
    }
};

export default theme;
