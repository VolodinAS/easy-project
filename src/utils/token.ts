import { localStorageTokenKey } from '../data/constants';

export const authToken = {
    get token() {
        return localStorage.getItem(localStorageTokenKey);
    },
    set token(token) {
        localStorage.setItem(localStorageTokenKey, token);
    },
    checkToken() {
        return Boolean(localStorage.getItem(localStorageTokenKey));
    },
    clearToken() {
        localStorage.removeItem(localStorageTokenKey);
    }
};