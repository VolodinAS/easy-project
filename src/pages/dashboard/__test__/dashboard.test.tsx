/* eslint-disable check-file/folder-naming-convention */
import React from 'react';
import { describe, it, expect, jest as jestGlobal, beforeAll, afterAll } from '@jest/globals';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { StaticRouter } from 'react-router-dom/server';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import 'whatwg-fetch';
import DashboardRoutes from '../../../dashboard-routes';

import theme from '../../../theme';
import dataProjectListSuccess from '../../../../stubs/json/dashboard/projects-list/success.json';

import { Provider } from 'react-redux';
import { store } from '../../../data/store';
import { authToken } from '../../../utils/token';

// jestGlobal.mock('react-router-dom', () => {

//     const originalModule = jest.requireActual('react-router-dom') as object;

//     const useOutletContext = () => [null, jest.fn()];

//     return {
//         __esModule: true,
//         ...originalModule,
//         useOutletContext
//     };
// });

const serverSuccess = setupServer(
    rest.get('*/projects', (req, res, ctx) => {
        return res(
            ctx.json(dataProjectListSuccess)
        );
    })
);

describe('test common Dashboard', () => {
    beforeAll( () => {
        serverSuccess.listen();
    } );
    afterAll( () => {
        serverSuccess.close();
    } );
    it('check Dashboard', async () => {

        authToken.token = 'token';

        const fieldRequired = 'My second project [можно нажимать]';
        const { container, getAllByText, getByTestId } = render(
            <StaticRouter location={'/easy-project/dashboard'}>
                <Provider store={store}>
                    <ThemeProvider theme={theme}>
                        <DashboardRoutes/>
                    </ThemeProvider>
                </Provider>
            </StaticRouter>
        );

        await waitFor(() => {
            const [labelChanged] = getAllByText(fieldRequired);
            expect(labelChanged.innerHTML).toContain(fieldRequired);

            const buttonDelete = getAllByText('Удалить')[1];

            fireEvent.click(buttonDelete);

        }, { timeout: 15000 });

        expect(container).toMatchSnapshot();

        await waitFor(() => {
            // const overlayData = screen.getByText(/Да/i);
            const overlayData = getByTestId('button-delete');
            expect(overlayData).toMatchSnapshot();
        }, { timeout: 15000 });

        
    });
});