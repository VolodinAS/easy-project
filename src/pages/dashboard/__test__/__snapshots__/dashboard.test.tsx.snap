// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test common Dashboard check Dashboard 1`] = `
<div>
  <div
    class="css-1boxdi0"
  >
    <div
      class="css-aoi44d"
    >
      <div
        class="css-174g2o2"
      >
        <a
          class="css-13lnaem"
          href="/easy-project/dashboard"
          title="Дашборд"
        >
          <div
            class="css-1povu0j"
          >
            <div
              class="css-4ucq2w"
            >
              <img
                alt="Easy project logo"
                src="test-file-stub"
              />
            </div>
            <p
              class="css-1vxbtp5"
            >
              <span>
                easy
              </span>
              <span>
                project
              </span>
            </p>
          </div>
        </a>
        <nav
          class="css-83k1q7"
        >
          <div
            class="css-16uff1u"
          >
            <span
              class="css-1c4ingb"
            >
              Дашборд
            </span>
          </div>
          <div
            class="css-16uff1u"
          >
            <button
              class="css-1640pth"
              title="language select"
              value="ru"
            >
              <img
                class="css-zlc08d"
                src="test-file-stub"
              />
            </button>
            <div
              class="css-746kv5"
            />
            <a
              class="active css-13xx20t"
              href="/easy-project/dashboard"
              title="Выход"
            >
              Выход
            </a>
          </div>
        </nav>
      </div>
    </div>
    <div
      class="css-14f5air"
    >
      <main
        class="css-mx63p3"
      >
        <div
          class="css-1iay1j5"
        >
          <div>
            <div
              class="css-q7ennu"
              id="overlay"
            >
              <div
                class="css-fslfiw"
              >
                <div
                  class="css-9rseyg"
                >
                  <button
                    class="css-4jvx38"
                  >
                    <img
                      alt="Закрыть"
                      src="test-file-stub"
                    />
                  </button>
                </div>
                <div
                  class="css-19jvvxe"
                >
                  <p
                    class="css-ovqjzh"
                  >
                    Вы действительно хотите удалить данный проект?
                  </p>
                </div>
                <div
                  class="css-106mbhu"
                >
                  <button
                    class="css-a5dza1"
                  >
                    Нет
                  </button>
                  <button
                    class="css-a5dza1"
                    data-testid="button-delete"
                  >
                    Да
                  </button>
                </div>
              </div>
            </div>
            <div
              class="css-44i7do"
            >
              <div
                class="css-3l4y0u"
              />
              <h1
                class="css-1s264hv"
              >
                Дашборд
              </h1>
            </div>
            <div
              class="css-1p9zq55"
            >
              <div
                class="css-vokxg9"
              >
                <div
                  class="css-ibdtyj"
                >
                  <div
                    class="css-1su66zo"
                  >
                    <p
                      class="css-10dvk2y"
                    >
                      Hello, world! [для стенда, не нажимать]
                    </p>
                  </div>
                  <div
                    class="css-ca8u3u"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Автор: testUser
                    </p>
                  </div>
                  <div
                    class="css-16qm4d5"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Создан: 19-04-2022
                    </p>
                  </div>
                  <div
                    class="css-16qm4d5"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Изменен: 19-04-2022
                    </p>
                  </div>
                </div>
                <div
                  class="css-1bn5kge"
                >
                  <a
                    class="css-1ky20yb"
                    href="/easy-project/dashboard/1"
                  >
                    <div
                      class="css-1x6z4y4"
                    >
                      <img
                        alt="Задачи"
                        src="test-file-stub"
                      />
                      <p
                        class="css-1l5tvrx"
                      >
                        Задачи
                      </p>
                    </div>
                  </a>
                  <a
                    class="css-1ky20yb"
                    href="/easy-project/dashboard/edit-project/1"
                  >
                    <div
                      class="css-1x6z4y4"
                    >
                      <img
                        alt="Редактировать"
                        src="test-file-stub"
                      />
                      <p
                        class="css-1l5tvrx"
                      >
                        Редактировать
                      </p>
                    </div>
                  </a>
                  <button
                    class="css-1exh3kr"
                    title="Удалить"
                  >
                    <img
                      alt="Удалить"
                      src="test-file-stub"
                    />
                    <p
                      class="css-1l5tvrx"
                    >
                      Удалить
                    </p>
                  </button>
                </div>
              </div>
              <div
                class="css-vokxg9"
              >
                <div
                  class="css-ibdtyj"
                >
                  <div
                    class="css-1su66zo"
                  >
                    <p
                      class="css-10dvk2y"
                    >
                      My second project [можно нажимать]
                    </p>
                  </div>
                  <div
                    class="css-ca8u3u"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Автор: testUser
                    </p>
                  </div>
                  <div
                    class="css-16qm4d5"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Создан: 19-04-2022
                    </p>
                  </div>
                  <div
                    class="css-16qm4d5"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Изменен: 19-04-2022
                    </p>
                  </div>
                </div>
                <div
                  class="css-1bn5kge"
                >
                  <a
                    class="css-1ky20yb"
                    href="/easy-project/dashboard/2"
                  >
                    <div
                      class="css-1x6z4y4"
                    >
                      <img
                        alt="Задачи"
                        src="test-file-stub"
                      />
                      <p
                        class="css-1l5tvrx"
                      >
                        Задачи
                      </p>
                    </div>
                  </a>
                  <a
                    class="css-1ky20yb"
                    href="/easy-project/dashboard/edit-project/2"
                  >
                    <div
                      class="css-1x6z4y4"
                    >
                      <img
                        alt="Редактировать"
                        src="test-file-stub"
                      />
                      <p
                        class="css-1l5tvrx"
                      >
                        Редактировать
                      </p>
                    </div>
                  </a>
                  <button
                    class="css-1exh3kr"
                    title="Удалить"
                  >
                    <img
                      alt="Удалить"
                      src="test-file-stub"
                    />
                    <p
                      class="css-1l5tvrx"
                    >
                      Удалить
                    </p>
                  </button>
                </div>
              </div>
              <div
                class="css-vokxg9"
              >
                <div
                  class="css-ibdtyj"
                >
                  <div
                    class="css-1su66zo"
                  >
                    <p
                      class="css-10dvk2y"
                    >
                      Final project [для стенда, не нажимать]
                    </p>
                  </div>
                  <div
                    class="css-ca8u3u"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Автор: testUser
                    </p>
                  </div>
                  <div
                    class="css-16qm4d5"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Создан: 19-04-2022
                    </p>
                  </div>
                  <div
                    class="css-16qm4d5"
                  >
                    <p
                      class="css-1lvqvn4"
                    >
                      Изменен: 19-04-2022
                    </p>
                  </div>
                </div>
                <div
                  class="css-1bn5kge"
                >
                  <a
                    class="css-1ky20yb"
                    href="/easy-project/dashboard/3"
                  >
                    <div
                      class="css-1x6z4y4"
                    >
                      <img
                        alt="Задачи"
                        src="test-file-stub"
                      />
                      <p
                        class="css-1l5tvrx"
                      >
                        Задачи
                      </p>
                    </div>
                  </a>
                  <a
                    class="css-1ky20yb"
                    href="/easy-project/dashboard/edit-project/3"
                  >
                    <div
                      class="css-1x6z4y4"
                    >
                      <img
                        alt="Редактировать"
                        src="test-file-stub"
                      />
                      <p
                        class="css-1l5tvrx"
                      >
                        Редактировать
                      </p>
                    </div>
                  </a>
                  <button
                    class="css-1exh3kr"
                    title="Удалить"
                  >
                    <img
                      alt="Удалить"
                      src="test-file-stub"
                    />
                    <p
                      class="css-1l5tvrx"
                    >
                      Удалить
                    </p>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-i9ol1x"
          >
            <a
              buttontheme="[object Object]"
              class="css-vkvbx5"
              href="/easy-project/dashboard/new-project"
              title="Дашборд"
            >
              Новый проект
            </a>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`test common Dashboard check Dashboard 2`] = `
<button
  class="css-a5dza1"
  data-testid="button-delete"
>
  Да
</button>
`;
